### ISSUE TITLE
NameError: name 'sageattention' is not defined

### ISSUE THREAD

**Issue by Gerkinfeltser** [blepping/ComfyUI-bleh#11]
Started getting the following at comfy startup (& bleh fails to load because of it)
```
Traceback (most recent call last):
  File "I:\ai\comfy\ComfyUI_windows_portable\ComfyUI\nodes.py", line 2026, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "I:\ai\comfy\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-bleh\__init__.py", line 10, in <module>
    from .py.nodes import (
  File "I:\ai\comfy\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-bleh\py\nodes\sageAttention.py", line 30, in <module>
    sageattn_function=sageattention.sageattn,
                      ^^^^^^^^^^^^^
NameError: name 'sageattention' is not defined
```

**Comment by blepping** [comment]
Sorry about that! I did something dumb while refactoring, it should be fixed now. Please update and try again.

Apologies for the inconvenience and thank you for the bug report!

**Comment by ronan36880** [comment]
Issue still happens for me as of commit acf99d639568ca85615b4945b2a8295d8d0973ec.

The following diff fixed it for me, but it's a workaround since it doesn't seem be the intended fix (variable is never set to something, in this case).

```diff
diff --git a/py/nodes/sageAttention.py b/py/nodes/sageAttention.py
index 753fa34..e783f8a 100644
--- a/py/nodes/sageAttention.py
+++ b/py/nodes/sageAttention.py
@@ -27,7 +27,7 @@ def attention_sage(  # noqa: PLR0917
     attn_precision=None,
     skip_reshape=False,
     sageattn_allow_head_sizes: None | set | tuple | list = None,
-    sageattn_function=sageattention.sageattn if sageattention is not None else None,
+    sageattn_function=sageattention.sageattn if sageattn is not None else None,
     sageattn_verbose=False,
     **kwargs: dict[str],
 ):
```

Had to rename `sageattention` in the condition to `sageattn`, as that's the variable that seems to be set above. I'm guessing the idea was to have python resolve `sageattention.sageattn` to the variable set to `None` in the same file, but it's not doing that for me.

<details>
<summary>Traceback</summary>

```
[2024-11-23 10:57:41.874] Traceback (most recent call last):
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\nodes.py", line 2027, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-bleh\__init__.py", line 10, in <module>
    from .py.nodes import (
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-bleh\py\nodes\ops.py", line 17, in <module>
    sonar_noise = importlib.import_module("custom_nodes.ComfyUI-sonar.py.noise")
  File "importlib\__init__.py", line 126, in import_module
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-sonar\__init__.py", line 1, in <module>
    from .py import freeu_extreme, nodes, powernoise, sonar
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-sonar\py\freeu_extreme.py", line 5, in <module>
    from .external import MODULES as EXTERNAL_MODULES
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-sonar\py\external.py", line 7, in <module>
    bleh = importlib.import_module("custom_nodes.ComfyUI-bleh")
  File "importlib\__init__.py", line 126, in import_module
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-bleh\__init__.py", line 10, in <module>
    from .py.nodes import (
  File "D:\Software\StabilityMatrix\Data\Packages\ComfyUI\custom_nodes\ComfyUI-bleh\py\nodes\sageAttention.py", line 30, in <module>
    sageattn_function=sageattention.sageattn if sageattention is not None else None,
NameError: name 'sageattention' is not defined. Did you mean: 'orig_attention'?
```

</details>

**Comment by Gerkinfeltser** [comment]
Success! Thank you!
& sorry for the misleading/premature Emoji! I should have done my due diligence & reacted only after I confirmed it was working. 😬

**Comment by blepping** [comment]
> sorry for the misleading/premature Emoji!

Definitely nothing to apologize for there! I just misinterpreted it. Anyway, glad to hear the issue is now resolved so I'll close this now.

If you or anyone else has any other problems, please feel free to create another issue. (Also discussions are open so just asking questions about stuff is fine also.)
